gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigV2ResourceLocalHandler.java

```diff
@@ -81,7 +83,7 @@ public CreateKVResponse createFlowConfig(FlowConfig flowConfig, boolean triggerL
     } else if (Boolean.parseBoolean(responseMap.getOrDefault(ServiceConfigKeys.COMPILATION_SUCCESSFUL, new AddSpecResponse<>("false")).getValue().toString())) {
       httpStatus = HttpStatus.S_201_CREATED;
     } else {
-      httpStatus = HttpStatus.S_400_BAD_REQUEST;
+      throw new RestLiServiceException(HttpStatus.S_400_BAD_REQUEST, "Flow was not compiled successfully");
```

Comment:
What are the next steps for the user here? Should he check the flow properties (we can point to the list of allowed properties), should he file a ticket and ask GaaS engineers to fix the issue? Also, do we have additional details on why the compilation was unsuccessful and what specific place had a problem?

Reply:
The challenge is that if a flow doesn't compile it's hard to know why because we don't know what was the intended path. It could be because of missing edges/nodes, or unresolved config keys. It's something that could be improved upon still, but most users of gaas self-serve should be using edges we've already defined so probably shouldn't have this problem anyway.

Reply:
So this will be primarily used by SREs? I can think of a situation when they have a 2-page flow and a typo in it. How would they find where the typo is? Are there any helpful compilation exceptions we can provide? For example: "problem is in the line 123, variable 'sssschema' is not found", or "no path found between 'faro' and 'warr', check the names of the source and the target").

If we can't pinpoint a single error and the problems are all around, we can send users to the compilation logs, and provide them some unique token so that they can find the relevant lines easily. For example: "for more information, check the GaaS logs that have requestId=XXX"

Reply:
The reason could be a couple things:

1. No path found. This could be due to: the edges don't have the paths they are using, there are no edges between those nodes, the nodes or edges were not added on gaas startup due to an error, etc.

2. There is an unresolved config. We do have a field that stores whatever errors we encountered with unresolved config.

The problems are:
- If it fails due to 1, there isn't really any way to know exactly what went wrong since we don't know what the intended path was.
- Even if it fails due to 1, we may have encountered some unresolved config on a separate unintended path. So it would be pretty confusing to include an error that some config was unresolved when the real reason was that the path.

If anything maybe the best message for now would just be "This could be because no path is found, or because of X unresolved config"?

Reply:
That can be a good start.